cmd/mirror-main.go

```diff
@@ -681,7 +681,7 @@ func runMirror(srcURL, dstURL string, ctx *cli.Context, encKeyDB map[string][]pr
 		fatalIf(errDummy(), "Synchronizing bucket policies is only possible when both source & target point to S3 servers.")
 	}
 
-	mirrorAllBuckets := (srcClt.GetURL().Type == objectStorage && srcClt.GetURL().Path == "/") ||
+	mirrorAllBuckets := (srcClt.GetURL().Type == objectStorage && srcClt.GetURL().Path == "/") &&
```

Comment:
why is this changing?

Reply:
Earlier had the perception that `mirrorAllBuckets` will be true only when alias are passed for both source and destination i.e. `mc mirror  minioA minioB` . But this was wrong so reverted it. 

In case `mirrorAllBuckets` is false i.e. `./mc mirror -a minioA/src minioB/src`  ( Only specified  buckets  are mirrored ) This case was failing earlier so added the else block in the code now.